AppTabs dropdown panics when pressed after second tab select

<!-- Please search for open issues that relate to the same problem before opening a new one. -->

### Describe the bug:
<!-- A clear and concise description about the bug. -->
Pressing the tab dropdown a second time causes fyne to panic over an invalid memory address. Pressing it the first time and selecting a tab works fine, the second time as well, but the third press panics. This does not seem to be an issue with DocTabs as far as I can see.

The trace of the crash:
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0xa8 pc=0x6e802e]

goroutine 22 [running]:
fyne.io/fyne/v2/internal/widget.(*overlayRenderer).MinSize(0xc0003c07e0, 0xc0003b97a0)
	/home/jacob/Git/fyne/internal/widget/overlay_container.go:124 +0x2e
fyne.io/fyne/v2/internal/widget.MinSizeOf(0xa46858, 0xc0003b97a0, 0xc0003b9140)
	/home/jacob/Git/fyne/internal/widget/base.go:69 +0x4f
fyne.io/fyne/v2/internal/widget.(*OverlayContainer).MinSize(...)
	/home/jacob/Git/fyne/internal/widget/overlay_container.go:49
fyne.io/fyne/v2/widget.NewPopUpMenu(0xc000063e78, 0x0, 0x0, 0x0)
	/home/jacob/Git/fyne/widget/popup_menu.go:23 +0x1a5
fyne.io/fyne/v2/container.buildPopUpMenu(0xa44538, 0xc0000b62d0, 0xc0008b23c0, 0xc000315e00, 0x5, 0x5, 0x0)
	/home/jacob/Git/fyne/container/tabs.go:91 +0xdd
fyne.io/fyne/v2/container.(*appTabsRenderer).buildOverflowTabsButton.func1()
	/home/jacob/Git/fyne/container/apptabs.go:309 +0x2bb
fyne.io/fyne/v2/widget.(*Button).Tapped(0xc0008b23c0, 0xc00039fbb0)
	/home/jacob/Git/fyne/widget/button.go:178 +0x74
fyne.io/fyne/v2/internal/driver/glfw.(*window).mouseClickedHandleTapDoubleTap.func1()
	/home/jacob/Git/fyne/internal/driver/glfw/window.go:873 +0x38
fyne.io/fyne/v2/internal/driver/common.(*Window).RunEventQueue(0xc0000c4000)
	/home/jacob/Git/fyne/internal/driver/common/window.go:58 +0x66
created by fyne.io/fyne/v2/internal/driver/glfw.(*gLDriver).createWindow.func1
	/home/jacob/Git/fyne/internal/driver/glfw/window.go:1372 +0x11c
```

### To Reproduce:
Steps to reproduce the behaviour:
1. Go to the AppTabs section in `cmd/fyne_demo`.
2. Click on the tabs dropdown and select as tab (x2, repeat twice)
3. Press the tab dropdown again
4. The application panics

### Screenshots:
<!-- If applicable, add screenshots or gifs to help explain your problem. -->
![apptabs-dropdown-panic](https://user-images.githubusercontent.com/25466657/120352792-6f064e80-c301-11eb-843e-ad01f62c4a3e.gif)

### Example code:
Just use `cmd/fyne_demo` in this situation.


### Device (please complete the following information):
 - **OS:** Linux
 - **Version:** Fedora 34 Workstation
 - **Go version:** 1.16.4
 - **Fyne version:** <!-- [e.g. 1.4.2 or git SHA] --> develop (v2.1.0) @ https://github.com/fyne-io/fyne/commit/12d020ec2dd30646a9d2a7944edd97298c7d3d28

--- a/container/apptabs.go
+++ b/container/apptabs.go
@@ -57,6 +57,8 @@ func (t *AppTabs) CreateRenderer() fyne.WidgetRenderer {
 		},
 		appTabs: t,
 	}
+	r.action = r.buildOverflowTabsButton()
+
 	// Initially setup the tab bar to only show one tab, all others will be in overflow.
 	// When the widget is laid out, and we know the size, the tab bar will be updated to show as many as can fit.
 	r.updateTabs(1)
@@ -407,13 +409,12 @@ func (r *appTabsRenderer) updateTabs(max int) {
 
 	// Set overflow action
 	if tabCount <= max {
-		r.action = nil
+		r.action.Hide()
 		r.bar.Layout = layout.NewMaxLayout()
 	} else {
 		tabCount = max
-		if r.action == nil {
-			r.action = r.buildOverflowTabsButton()
-		}
+		r.action.Show()
+
 		// Set layout of tab bar containing tab buttons and overflow action
 		if r.appTabs.location == TabLocationLeading || r.appTabs.location == TabLocationTrailing {
 			r.bar.Layout = layout.NewBorderLayout(nil, r.action, nil, nil)

